banking_system.rb

```diff
@@ -0,0 +1,133 @@
+class Account
+	@@number = 23456789101
+	@@balance = 0
+	def initialize
+		@acc_number = @@number
+	end
+
+	def open_account
+		@@number += 1
+		@acc_number = @@number
+	end
+
+	def deposit_money(amount)
+		@@balance += amount
+	end
+
+	def withdraw_money(amount)
+		if @@balance <= 0 || @@balance <= amount
+			return false
+		else
+			@@balance -= amount
+			return true
+		end
+	end
```

Comment:
This logic is flawed as every person is sharing same balance even they did not deposit the money.